### PR TITLE
Added more documentation for dealing with big datasets

### PR DESCRIPTION
**Pull request by freemansw1**
Added more details to our documentation around dealing with big datasets, including a new example notebook of running feature detection in parallel.

With @RBhupi from tobathon February 2024

* [x] Have you followed our guidelines in CONTRIBUTING.md? 
* [x] Have you self-reviewed your code and corrected any misspellings? 
* [x] Have you written documentation that is easy to understand?
* [x] Have you written descriptive commit messages? 
* [ ] Have you added NumPy docstrings for newly added functions? 
* [ ] Have you formatted your code using black? 
* [ ] If you have introduced a new functionality, have you added adequate unit tests?
* [x] Have all tests passed in your local clone? 
* [x] If you have introduced a new functionality, have you added an example notebook?
* [x] Have you kept your pull request small and limited so that it is easy to review? 
* [x] Have the newest changes from this branch been merged?

### CODE REVIEW

**Review by w-k-jones — COMMENTED** [review]
Looks good. The only change I would suggest to the notebook is performing tracking to show that this needs to be performed on a single dataframe, but that it can be done without loading all spatial data into memory. The notebook isn't yet displayed in the docs, is that intended? And the failure of the jupyter notebooks CI can be fixed with the inclusion of dask in `example_requirements.txt`, which has been done in #334

**Comment by freemansw1** [comment]
I'm ready for a review!

**Review by w-k-jones — APPROVED** [review]
All looks good to me! Happy for this to be merged when ready

File: doc/big_datasets_examples/index.rst
Comment (fsenf) [inline doc/big_datasets_examples/index.rst:1]: don't think that the index.rst is needed at all

File: doc/big_datasets_examples/notebooks/parallel_processing_tobac.ipynb
Comment (fsenf) [inline doc/big_datasets_examples/notebooks/parallel_processing_tobac.ipynb:1]: Please, resubmit this notebook after formatted with black!

File: doc/big_datasets.rst [hunk 1]
```diff
@@ -1,11 +1,42 @@
...
+Current versions of threshold feature detection (see :doc:`feature_detection_overview`) are time independent, meaning that one can easily parallelize feature detection across all times (although not across space). *tobac* provides the :py:meth:`tobac.utils.combine_feature_dataframes` function to combine a list of dataframes produced by a parallelization method (such as :code:`jug`,  :code:`multiprocessing.pool`, or :code:`dask.bag`) into a single combined dataframe suitable to perform tracking with.
+
+Below is a snippet from a larger notebook demonstrating how to run feature detection in parallel ( :doc:`big_datasets_examples/notebooks/parallel_processing_tobac`).
+
+.. code-block:: python
+
+   # build list of tracked variables using Dask.Bag
+   b = dask.bag.from_sequence([combined_ds['data'][x:x+1] for x in range(len(combined_ds['time']))], npartitions=1)
+   out_feature_dfs = dask.bag.map(lambda x: tobac.feature_detection_multithreshold(x.to_iris(), 4000, **parameters_features), b).compute()
+   combined_dataframes= tobac.utils.general.combine_feature_dataframes(out_feature_dfs)
```
Comment (fsenf) [inline doc/big_datasets.rst:20]: Please care that the code snippet is also formatted with black (maybe you copy it over once the notebook has been formatted,

**Review by fsenf — CHANGES_REQUESTED** [review]
This is a nice addition. Thank you for your work!

I requested some formatting changes - which will be easy to do

File: doc/big_datasets_examples/notebooks/parallel_processing_tobac.ipynb
Comment (w-k-jones) [reply doc/big_datasets_examples/notebooks/parallel_processing_tobac.ipynb:1]: We should apply black formatting to all the notebooks at some point...

Comment (freemansw1) [reply doc/big_datasets_examples/notebooks/parallel_processing_tobac.ipynb:1]: I went ahead and formatted all of the notebooks with black while I was at it

File: doc/big_datasets_examples/index.rst
Comment (freemansw1) [reply doc/big_datasets_examples/index.rst:1]: It was to display the way I had it, but I moved it to the notebook gallery instead.

**Comment by freemansw1** [comment]
Ok, @fsenf, I believe I've addressed your comments and I am now ready for a re-review.

**Review by fsenf — APPROVED** [review]
Thank you for update! I am happy with the changes. I especially like that the doc notebooks have been integrated into the example gallery: https://tobac--408.org.readthedocs.build/en/408/examples.html

Please go ahead and merge!

Cheers, Fabian.

PS: It would be cool to have nice looking thumbnails for the new gallery entry. I will create a new issue for that request.

**Pull request merged**